sw/tests/axirt_budget.c

```diff
@@ -3,60 +3,101 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 // Thomas Benz <tbenz@iis.ee.ethz.ch>
+// Alessandro Ottaviano <aottaviano@iis.ee.ethz.ch>
 //
 // Validate the budget functionality of AXI RT
 
 #include "axirt.h"
 #include "dif/dma.h"
-#include "regs/axi_rt.h"
 #include "params.h"
+#include "regs/axi_rt.h"
+#include "regs/cheshire.h"
 #include "util.h"
 
 // transfer
-#define SIZE_BYTES 256
-#define SRC_STRIDE 0
-#define DST_STRIDE 0
-#define NUM_REPS 8
-#define SRC_ADDR 0x0000000010000000
-#define DST_ADDR 0x0000000080000000
+#define SIZE_BEAT_BYTES 8
+#define DMA_NUM_BEATS 32
+#define DMA_NUM_REPS 8
+#define DMA_SIZE_BYTES (SIZE_BEAT_BYTES * DMA_NUM_BEATS)
+#define DMA_TOTAL_SIZE_BYTES (DMA_SIZE_BYTES * DMA_NUM_REPS)
+#define DMA_SRC_STRIDE 0
+#define DMA_DST_STRIDE 0
+#define SRC_ADDR 0x0000000078000000 // L2
```

Comment:
On closer Inspection: That's not where the SPM is in Cheshire :sweat_smile: We should double-check the tests.

Reply:
Fixed, now the code fails correctly on wrong addresses. Thanks for spotting this! Somehow it gost lost in translation when porting the fixed test from Carfield